types/agilite/index.d.ts

```diff
@@ -0,0 +1,227 @@
+// Type definitions for agilite 6.4
+// Project: https://www.npmjs.com/package/agilite
+// Definitions by: Armand Smit <https://github.com/ArrieAgilite>
+//                 John Jardin <https://github.com/johnjardin>
+//                 Agilit-e <https://github.com/agilitehub>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+export class Agilite {
```

Comment:
The JavaScript library is a CommonJS module, but these definitions are for an ES2015 module. `module.exports =` in JavaScript needs to be declared as `export =` in .d.ts files. To export your support types from here, you’ll need to put them in a namespace also named `Agilite`:

```ts
declare class Agilite {
  // ...
}

declare namespace Agilite {
  interface AgiliteConfig {
    // ...
  }
  interface ReqType {
    // ...
  }
  interface Sort {
    // ...
  }
  // ...
}

export = Agilite;
```